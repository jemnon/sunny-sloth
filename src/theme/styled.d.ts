import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
