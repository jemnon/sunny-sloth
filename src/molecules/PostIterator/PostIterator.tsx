import React, { FC, ReactNode } from 'react';
import styled from 'styled-components';
import { up } from 'styled-breakpoints';
import { ButtonReset } from '../../atoms/Button';
import ImgWrapper from '../../atoms/ImgWrapper';
import Text from '../../atoms/Text';

interface PostIteratorProps {
  prev?: {
    image?: ReactNode;
    name: string;
    slug: string;
  };
  next?: {
    image?: ReactNode;
    name: string;
    slug: string;
  };
  onClick: (slug: string) => void;
}

const PostIteratorContainer = styled.section`
  display: flex;
  justify-content: space-between;

  width: 100%;

  padding: ${({ theme }): string => theme.spacing.sm4};

  background-color: ${({ theme }): string => theme.colors.nearWhite};
`;

const PostIteratorButton = styled.button`
  ${ButtonReset};

  display: flex;
  align-items: center;

  cursor: pointer;
`;

const PostIteratorFigure = styled.figure`
  width: 4rem;
  ${up('sm')} {
    width: 5rem;
  }
`;

const PostIteratorContent = styled.div<{ direction?: string }>`
  display: flex;
  flex-direction: column;

  ${({ direction = 'left', theme }): string => {
    return direction === 'right'
      ? `margin-right: ${theme.spacing.sm4}`
      : `margin-left: ${theme.spacing.sm4}`;
  }};
`;

const PostIteratorName = styled.div`
  display: none;
  ${up('sm')} {
    display: block;
  }
`;

const PostIterator: FC<PostIteratorProps> = ({ next, prev, onClick }) => {
  const handleClick = (pathname: string): void => {
    onClick(pathname);
  };
  return (
    <PostIteratorContainer>
      {prev?.slug && (
        <PostIteratorButton onClick={(): void => handleClick(prev.slug)}>
          <PostIteratorFigure>
            <ImgWrapper ratio={1 / 1}>{prev.image}</ImgWrapper>
          </PostIteratorFigure>
          <PostIteratorContent>
            <Text textTransform="uppercase" fontSize="f-sm" fontWeight="bold">
              Previous
            </Text>
            <PostIteratorName
              as={Text}
              textTransform="uppercase"
              textColor="orange"
              fontSize="f-sm"
              fontWeight="bold"
            >
              {prev.name}
            </PostIteratorName>
          </PostIteratorContent>
        </PostIteratorButton>
      )}
      {next?.slug && (
        <PostIteratorButton onClick={(): void => handleClick(next.slug)}>
          <PostIteratorContent direction="right">
            <Text
              textAlign="right"
              textTransform="uppercase"
              fontSize="f-sm"
              fontWeight="bold"
            >
              Next
            </Text>
            <PostIteratorName
              as={Text}
              textTransform="uppercase"
              textColor="orange"
              fontSize="f-sm"
              fontWeight="bold"
            >
              {next.name}
            </PostIteratorName>
          </PostIteratorContent>
          <PostIteratorFigure>
            <ImgWrapper ratio={1 / 1}>{next.image}</ImgWrapper>
          </PostIteratorFigure>
        </PostIteratorButton>
      )}
    </PostIteratorContainer>
  );
};

export default PostIterator;
