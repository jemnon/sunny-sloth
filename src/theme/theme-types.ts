export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
