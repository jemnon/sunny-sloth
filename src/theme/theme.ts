const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
