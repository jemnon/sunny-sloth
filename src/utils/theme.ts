import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
