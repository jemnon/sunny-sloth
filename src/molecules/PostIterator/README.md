# PostIterator

### Description
Component description

```js
import PostIterator from '@fenderdigital/ui/FILL IN TYPEPostIterator';
```

### Props
* `propName` - property name and description 
