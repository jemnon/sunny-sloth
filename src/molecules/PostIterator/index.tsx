import PostIterator from './PostIterator';

export default PostIterator;