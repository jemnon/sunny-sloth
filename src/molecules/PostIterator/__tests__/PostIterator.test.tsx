import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import PostIterator from '../PostIterator';
import theme from '../../../theme';

describe('<PostIterator />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <PostIterator
        next={{
          image: <img src="some-image.jpg" />,
          name: 'Tomato Bisque',
          slug: 'tomato-bisque',
        }}
        prev={{
          image: <img src="some-image.jpg" />,
          name: 'Shrimp Scampi',
          slug: 'shrimp-scampi',
        }}
        onClick={jest.fn}
      />
    </ThemeProvider>,
  );
  it('exists', (): void => {
    expect(PostIterator).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
