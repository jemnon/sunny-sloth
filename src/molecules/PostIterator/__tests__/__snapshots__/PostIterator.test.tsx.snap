// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostIterator /> renders defaults correctly 1`] = `
<section
  className="sc-hKgILt gRxjzL"
>
  <button
    className="sc-eCssSg jLhGTw"
    onClick={[Function]}
  >
    <figure
      className="sc-jSgupP frXCPv"
    >
      <div
        className="sc-gsTCUz iaHCbG"
      >
        <img
          src="some-image.jpg"
        />
      </div>
    </figure>
    <div
      className="sc-gKsewC iyJKSp"
    >
      <div
        className="sc-dlfnbm gnOgrr"
        fontSize="f-sm"
        fontWeight="bold"
      >
        Previous
      </div>
      <div
        className="sc-dlfnbm eobeyX sc-iBPRYJ QuaPf"
        fontSize="f-sm"
        fontWeight="bold"
        title="Shrimp Scampi"
      >
        Shrimp Scampi
      </div>
    </div>
  </button>
  <button
    className="sc-eCssSg jLhGTw"
    onClick={[Function]}
  >
    <div
      className="sc-gKsewC ircwHW"
      direction="right"
    >
      <div
        className="sc-dlfnbm dUeyhs"
        fontSize="f-sm"
        fontWeight="bold"
      >
        Next
      </div>
      <div
        className="sc-dlfnbm eobeyX sc-iBPRYJ QuaPf"
        fontSize="f-sm"
        fontWeight="bold"
        title="Tomato Bisque"
      >
        Tomato Bisque
      </div>
    </div>
    <figure
      className="sc-jSgupP frXCPv"
    >
      <div
        className="sc-gsTCUz iaHCbG"
      >
        <img
          src="some-image.jpg"
        />
      </div>
    </figure>
  </button>
</section>
`;
